src/NUnitFramework/framework/Internal/TestCaseParameters.cs

```diff
@@ -31,7 +31,7 @@ namespace NUnit.Framework.Internal
     /// other selected parameters needed for constructing
     /// a parameterized test case.
     /// </summary>
-    public class TestCaseParameters : TestParameters, ITestCaseData, IApplyToTest
+    public class TestCaseParameters : TestParameters, ITestCaseData
```

Comment:
If the `ApplyToTest` method does the same thing that Attributes implementing the `IApplyToTest` interface do, then I like the idea that we signal this to the developers, even though an interface is not strictly needed.

Reply:
This is not possible unless we stop passing the TestNameGenerator to the method which requires either the Test*Parameters classes to know the default test name generator, or for us to go back to the builder asking for all name-related arguments rather than telling the parameters to apply the name (plus duplicating the name application logic between the two builders).

Reply:
I agree that the way the code is written currently is wrong. It's not how any other attribute or object works. The `TestCaseParameters` object should decide what to do in it's ApplyToTest method __if anything__ and the method should always be called. But ApplyToTest should not be responsible for the __default__ name. That should just be there before it is ever called.

It's true that doing it that way causes the name to be set twice, but that doesn't seem to have caused any problems up to now.

Reply:
Let me clarify: I'm not passing the default name generator in order to generate the default name. I'm passing the default name generator in order to use the default name pattern with *non-default* overridden argument display names.

Reply:
Ah! That makes a bit more sense. My assumption was that `SetArgNames` would be just a wrapper around `SetName("{m}(whatever,you,tell,it)")`.

Reply:
Can't do that in case an argument contained curlies... there's no escape method =D

Reply:
To be honest I do think it's cleaner this way, plus `{m}{a:40}` works.

Reply:
I do actually believe I did the simplest thing possible. =D Could be wrong, but so far I think so.